src/main/java/org/threadly/concurrent/UnfairExecutor.java

```diff
@@ -0,0 +1,436 @@
+   * to using the {@link TaskHashXorTimeStripeGenerator}.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param useDaemonThreads {@code true} if created threads should be daemon
+   */
+  public UnfairExecutor(int threadCount, boolean useDaemonThreads) {
+    this(threadCount, useDaemonThreads, TaskHashXorTimeStripeGenerator.instance());
+  }
+
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count.  
+   * 
+   * Possible built in stripe generators for use would be {@link AtomicStripeGenerator} or 
+   * {@link TaskHashXorTimeStripeGenerator}.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param useDaemonThreads {@code true} if created threads should be daemon
+   * @param stripeGenerator Generator for figuring out how a task is assigned to a thread
+   */
+  public UnfairExecutor(int threadCount, boolean useDaemonThreads, 
+                        TaskStripeGenerator stripeGenerator) {
+    this(threadCount, 
+         new ConfigurableThreadFactory(UnfairExecutor.class.getSimpleName() + "-", true, 
+                                       useDaemonThreads, Thread.NORM_PRIORITY, null, null), 
+         stripeGenerator);
+  }
+
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count and factory.  This also 
+   * defaults to using the {@link TaskHashXorTimeStripeGenerator}.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param threadFactory thread factory for producing new threads within executor
+   */
+  public UnfairExecutor(int threadCount, ThreadFactory threadFactory) {
+    this(threadCount, threadFactory, TaskHashXorTimeStripeGenerator.instance());
+  }
+
+  /**
+   * Constructs a new {@link UnfairExecutor} with a provided thread count and factory.  
+   * 
+   * Possible built in stripe generators for use would be {@link AtomicStripeGenerator} or 
+   * {@link TaskHashXorTimeStripeGenerator}.
+   * 
+   * @param threadCount Number of threads, recommended to be a prime number
+   * @param threadFactory thread factory for producing new threads within executor
+   * @param stripeGenerator Generator for figuring out how a task is assigned to a thread
+   */
+  public UnfairExecutor(int threadCount, ThreadFactory threadFactory, 
+                        TaskStripeGenerator stripeGenerator) {
+    ArgumentVerifier.assertGreaterThanZero(threadCount, "threadCount");
+    ArgumentVerifier.assertNotNull(stripeGenerator, "stripeGenerator");
+    
+    this.schedulers = new Worker[threadCount];
+    this.shutdownStarted = new AtomicBoolean(false);
+    this.stripeGenerator = stripeGenerator;
+    
+    for (int i = 0; i < threadCount; i++) {
+      schedulers[i] = new Worker(threadFactory);
+      if (i > 0) {
+        schedulers[i].setNeighborWorker(schedulers[i - 1]);
+      }
+    }
+    schedulers[0].setNeighborWorker(schedulers[schedulers.length - 1]);
```

Comment:
Honestly this makes no sense for a thread count of one.  While it is technically possible, "SingleThreadScheduler" would be a much better (faster) and more capable option.

I considered even throwing an exception if you try to do this in a single threaded way, but figured it technically works, so why not.

In any case, we don't want special logic here for the single thread case, because if we do then we must null check in the execution loop, which I obviously don't want.  And while more expensive, it will technically work.